### PR TITLE
sescan: prioritize scanning new pushes (PROJQUAY-3287)

### PR DESCRIPTION
**Pull request by syed**
Currently we randomly scan for entries to be indexed. This could take a
long time if we have a lot of unindexed images. Ideally, we want to
priortize scanning of newer images before we go to the old ones.

This PR creates a smaller batch of size `batch_size` from
`max_id - batch_size` to `max_id` to priortize scanning
those

### CODE REVIEW

**Comment by syed** [comment]
@kwestpharedhat  I didn't have permission to push to your fork so I closed your PR (#1145 ) and opened this one with your changes

File: data/secscan_model/secscan_v4_model.py [hunk 1]
```diff
@@ -202,11 +202,23 @@ def needs_reindexing_query(indexer_hash):
...
+            # Guarantees at least one batch of most recent manifests
             yield_random_entries(
                 not_indexed_query,
                 Manifest.id,
                 batch_size,
                 max_id,
+                # Because we have multiple workers and must stay random to ensure minimal duplicative work
+                # we will instead limit the set worked to the most recent batch.
+                max_id - batch_size,
+            ),
```
Comment (kleesc) [inline data/secscan_model/secscan_v4_model.py:214]: Won't this stop doing any work once `(max_id - batch_size) ... max_id` is done?

File: data/secscan_model/secscan_v4_model.py [hunk 2]
```diff
@@ -202,11 +202,23 @@ def needs_reindexing_query(indexer_hash):
...
+            # Could be a huge batch of work
+            yield_random_entries(
+                not_indexed_query,
+                Manifest.id,
+                batch_size,
+                # compensation for the above batch guarentee
+                max_id - batch_size,
```
Comment (kleesc) [inline data/secscan_model/secscan_v4_model.py:221]: Shouldn't this second iterator be left to `min_id ... max_id`. We still want to backfill the older manifests. Adding the iterator from `max_id - batch_size ... max_id` before this one should prioritize the new ones.

File: data/secscan_model/secscan_v4_model.py (hunk 1 above)
Comment (syed) [reply data/secscan_model/secscan_v4_model.py:214]: Yes but `max_id` keeps getting updated every time the worker does a `perform_indexing` call

File: data/secscan_model/secscan_v4_model.py (hunk 2 above)
Comment (syed) [reply data/secscan_model/secscan_v4_model.py:221]: The previous block was from `min_id -> max_id`  We are splitting it into two `min_id -> max_id - batch_size` and from `max_id - batch_size -> max_id` and the latter is the first one to be scanned

File: data/secscan_model/secscan_v4_model.py (hunk 1 above)
Comment (kleesc) [reply data/secscan_model/secscan_v4_model.py:214]: Isn't it the `min_id` that gets updated? 
i.e the `max_id` if the last batch becomes the new `min_id`

Comment (syed) [reply data/secscan_model/secscan_v4_model.py:214]: both get updated
```
        min_id = (
            start_token.min_id
            if start_token is not None
            else Manifest.select(fn.Min(Manifest.id)).scalar()
        )
        max_id = Manifest.select(fn.Max(Manifest.id)).scalar()
```

**Review by kleesc — APPROVED** [review]

**Pull request merged**